### PR TITLE
Further Save and restore hooks

### PR DESCRIPTION
**Pull request by ashb**
Further to #230 (and based on it, with the same initial commit keeping authorship) but extended to have more hook points.

Closes #230, #213

### CODE REVIEW

File: docs/hooks.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+- `@resurrect-hook-post-save-all`
+
+  Called at end of save process right before the spinner is turned off.
+
+- `@resurrect-hook-pre-restore-all`
+
+  Called before any tmux state is altered.
+
+- `eresurrect-hook-pre-restore-history`
```
Comment (bruno-) [inline docs/hooks.md]: Typo

File: docs/hooks.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  Called after panes and layout have been restores, but before bash history is
```
Comment (bruno-) [inline docs/hooks.md]: typo ... "have been restore**d**"

File: docs/hooks.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  restored (if it is enabled) -- the hook is always called even is history
```
Comment (bruno-) [inline docs/hooks.md]: **if** history ..

File: docs/hooks.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  saving is disabled.
+
+- `@resurrect-hook-pre-restore-pane-processes`
+
+  Called after history is resotred, but before running processes are restored.
```
Comment (bruno-) [inline docs/hooks.md]: "Called after history is **restored**"

**Comment by ashb** [comment]
@bruno- typos fixed, thanks!

**Pull request merged**